servicetalk-traffic-resilience-http/src/test/java/io/servicetalk/traffic/resilience/http/TrafficResilienceHttpServiceFilterTest.java

```diff
@@ -223,49 +223,53 @@ void testStopAcceptingConnections(final boolean dryRun, final String protocol) t
-                .protocols(protocolConfig)
-                .socketOption(CONNECT_TIMEOUT, (int) SECONDS.toMillis(CI ? 4 : 2))
-                .buildStreaming();
-
-        // First request -> Pending 1
-        final StreamingHttpRequest meta1 = client.newRequest(HttpRequestMethod.GET, "/");
-        client.reserveConnection(meta1)
-                .flatMap(it -> it.request(meta1))
-                .concat(Completable.defer(() -> {
-                    // First request, has a "never" pub as a body, we don't attempt to consume it.
-                    // Concat second request -> out of capacity -> server yielded
-                    final StreamingHttpRequest meta2 = client.newRequest(HttpRequestMethod.GET, "/");
-                    return client.reserveConnection(meta2).flatMap(it -> it.request(meta2)).ignoreElement();
-                }))
-                .toFuture()
-                .get();
+            try (StreamingHttpClient client = HttpClients.forSingleAddress(serverHostAndPort(serverContext))
+                    .protocols(protocolConfig)
+                    .socketOption(CONNECT_TIMEOUT, (int) SECONDS.toMillis(CI ? 4 : 2))
+                    .buildStreaming()) {
 
-        // Netty will evaluate the "yielding" (i.e., auto-read) on this attempt, so this connection will go through.
-        assertThat(client.reserveConnection(client.newRequest(HttpRequestMethod.GET, "/"))
-                .toFuture().get().asConnection(), instanceOf(HttpConnection.class));
+                // First request -> Pending 1
+                final StreamingHttpRequest meta1 = client.newRequest(HttpRequestMethod.GET, "/");
+                client.reserveConnection(meta1)
+                        .flatMap(it -> it.request(meta1))
+                        .concat(Completable.defer(() -> {
+                            // First request, has a "never" pub as a body, we don't attempt to consume it.
+                            // Concat second request -> out of capacity -> server yielded
+                            final StreamingHttpRequest meta2 = client.newRequest(HttpRequestMethod.GET, "/");
+                            return client.reserveConnection(meta2).flatMap(it -> it.request(meta2)).ignoreElement();
+                        }))
+                        .toFuture()
+                        .get();
 
-        // This connection shall full-fil the BACKLOG=1 setting
-        assertThat(client.reserveConnection(client.newRequest(HttpRequestMethod.GET, "/"))
-                .toFuture().get().asConnection(), instanceOf(HttpConnection.class));
+                // We expect up to a couple connections to succeed due to the intrinsic race between disabling accepts
+                // and new connect requests, as well as to account for kernel connect backlog. However, we do expect it
+                // to fail after a fairly short number of iterations.
+                for (int i = 0; i < 5; i++) {
```

Comment:
`5` -> should this be proportionate to `TCP_BACKLOG`? I read the commit message description but I'm assuming the test can sequence such that "we stop accepting connections" and "try to make a new connection" so the kernel should only accept `TCP_BACKLOG [+ 1]` connections before queues fill up bcz server isn't doing `accept`

Reply:
We can't enforce a happens-before wrt the 'stop accepting' and 'make a new one'. I've lowered the value to three, since that appears to be the upper bound for when we stop accepting connections.

Reply:
Where does 3 come from? Can you clarify the mechanics and who is "we" in this statement `when we stop accepting connections.`?

Reply:
3 comes from the existing behavior and justifications of the tests:
1 - satisfying a pending accept
2 - backlog of 1 for both linux and darwin (although params have different interpretations)
3 - This one should fail.
It happens that 2 also fails from time to time with the behavior that we expect from 3, and that is what results in the flakyness.

The "we" is the listener loop.

Reply:
would `TCP_BACKLOG + 2` make sense with comments similar to what you have above (potentially considering the off-by-one difference for linux)?

Reply:
I don't think we should base it on `TCP_BACKLOG` because that symbol has a different purpose than you'd expect (and thus it's big associated comment above the definition). The TLDR is that the backlog is effectively 1 across the board, but the value of `TCP_BACKLOG` _the symbol_ is OS dependent because unfortunately linux and darwin have different interpretations of the value.